File: easy/guess-number-higher-or-lower/guess-number-higher-or-lower.py
class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """

        l = 1
        r = n
        while l <= r:
            m = (l + r) // 2
            if guess(m) == 0:
                return m
            if guess(m) == -1:
                r = m - 1
            if guess(m) == 1:
                l = m + 1

File: easy/guess-number-higher-or-lower/guess-number-higher-or-lower.cpp

class Solution {
public:
    int guessNumber(int n) {
        int l = 1;
        int r = n;
        while (l <= r) {
            int m = l + (r - l) / 2;
            if (guess(m) == 0) {
                return m;
            }
            if (guess(m) == 1) {
                l = m + 1;
            }
            if (guess(m) == -1) {
                r = m - 1;
            }
        }
        return 0;
    }
};
